src/item.h

```diff
@@ -690,8 +690,10 @@ class item : public visitable<item>
          * @param insulation Amount of insulation item has from surroundings
          * @param pos The current position
          * @param carrier The current carrier
+         * @param flag is used to tell that the item is in a temperature controlled place. "heater", "fridge", "freezer", "root_cellar"
          */
-        void process_temperature_rot( int temp, float insulation, const tripoint pos, player *carrier );
+        void process_temperature_rot( int temp, float insulation, const tripoint pos, player *carrier,
```

Comment:
Only had time for a cursory review, but I think an enum with a default argument would make much more sense here than a string flag.

Reply:
I changed it to use enum. First time using them so that probably needs to be checked well.

process and process_temperature_rot default to temperature_flag::TEMP_NORMAL.